Client/core/Graphics/CPrimitiveMaterialBatcher.cpp

```diff
@@ -93,31 +92,15 @@ void CPrimitiveMaterialBatcher::UpdateMatrices(float fViewportSizeX, float fView
 }
 ////////////////////////////////////////////////////////////////
 //
-// CPrimitiveMaterialBatcher::Flush
+// CPrimitiveMaterialBatcher::SetDeviceStates
+//
 //
-// Send all buffered vertices to D3D
 //
 ////////////////////////////////////////////////////////////////
-void CPrimitiveMaterialBatcher::Flush(void)
+void CPrimitiveMaterialBatcher::SetDeviceStates()
 {
-    if (m_primitiveList.empty())
-        return;
-
-    // Save render states
-    IDirect3DStateBlock9* pSavedStateBlock = nullptr;
-    m_pDevice->CreateStateBlock(D3DSBT_ALL, &pSavedStateBlock);
-    // Set transformations
-    D3DXMATRIX matWorld;
-    D3DXMatrixIdentity(&matWorld);
-    m_pDevice->SetTransform(D3DTS_WORLD, &matWorld);
-    m_pDevice->SetTransform(D3DTS_VIEW, &m_MatView);
-    m_pDevice->SetTransform(D3DTS_PROJECTION, &m_MatProjection);
-
-    // Set vertex FVF
-    m_pDevice->SetFVF(PrimitiveMaterialVertice::FNV);
-
     // Set states
-    m_pDevice->SetRenderState(D3DRS_ZENABLE, m_bZTest ? D3DZB_TRUE : D3DZB_FALSE);
+    m_pDevice->SetRenderState(D3DRS_ZENABLE, D3DZB_TRUE);
```

Comment:
Depth-buffering is not required for 2D draws, so D3DRS_ZENABLE should be set to D3DZB_FALSE

Reply:
I was thinking about stuff like this: https://www.youtube.com/watch?v=CbJerQJBM0M where 2D images, or in this case primitives are placed in 3D world using shaders.

Reply:
By convention, the 2D drawing functions draw in front of the final world render (The video shows the logo always in front of the car). So the depth buffer should be
disabled by default.

For drawing primitives in the 3D world, we could add dxDrawPrimitive3D (similar to dxDrawLine3D)

Reply:
no `we could`, we must add 3d version

Reply:
Fair enough, disabled the depth buffering for material primitives :)